internal/core/software_renderer.rs

```diff
@@ -925,7 +926,7 @@ impl SharedBufferCommand {
 
 #[derive(Debug)]
 struct RoundedRectangle {
-    radius: PhysicalLength,
+    radius: PhysicalBorderRadius,
```

Comment:
Slightly sad about the increased memory usage here. But that's just how it is.

Reply:
What if we added one more item type alongside `Rectangle` and `RoundedRectangle`?

```rust
pub struct RoundedRectangle {
    pub background: Property<Brush>,
    pub border_width: Property<LogicalLength>,
    pub border_radius: Property<LogicalLength>,
    pub border_color: Property<Brush>,
    pub cached_rendering_data: CachedRenderingData,
}

impl Item for RoundedRectangle {
    fn render(
        self: Pin<&Self>,
        backend: &mut ItemRendererRef,
        self_rc: &ItemRc,
        size: LogicalSize,
    ) -> RenderingResult {
        (*backend).draw_rounded_rectangle(self, self_rc, size);
        RenderingResult::ContinueRenderingChildren
    }
}

pub struct BorderRectangle {
    pub background: Property<Brush>,
    pub border_width: Property<LogicalLength>,
    pub border_radius: Property<LogicalLength>,
    pub border_top_left_radius: Property<LogicalLength>,
    pub border_top_right_radius: Property<LogicalLength>,
    pub border_bottom_left_radius: Property<LogicalLength>,
    pub border_bottom_right_radius: Property<LogicalLength>,
    pub border_color: Property<Brush>,
    pub cached_rendering_data: CachedRenderingData,
}

impl Item for BorderRectangle {
    fn render(
        self: Pin<&Self>,
        backend: &mut ItemRendererRef,
        self_rc: &ItemRc,
        size: LogicalSize,
    ) -> RenderingResult {
        (*backend).draw_border_rectangle(self, self_rc, size);
        RenderingResult::ContinueRenderingChildren
    }
}
```